MAIN - updates buttons

paschalidi opened the pull request:
Happy to present to you 
our 
✨ **n e w**  ✨ **a m a z i n g** ✨ **b u t t o n s** ✨

#### On the details!
* `height` designers told me that would be 3 heights 56 40 and 32. Property `height` now can take three values `lg`,`sm`, `basic`(basic is the default value)
* restricted the already existing property `type` to: `type: PropTypes.oneOf(["primary", "secondary"])`
* instead of hardcoded hex colors this time we use our own `Colors`!
* wrote cool stories about all the different cases

#### g i f s  -  w o w 
![anchor-buttons](https://user-images.githubusercontent.com/4181674/56357073-e2f54380-61da-11e9-835d-e615c7778898.gif)
![buttons](https://user-images.githubusercontent.com/4181674/56357075-e38dda00-61da-11e9-8472-de5910ecf9e3.gif)

paschalidi committed:
adds new styles and components that are affected by the changes


styles


imageshots


with text this time

paschalidi committed:
shots

paschalidi committed:
Anchor

paschalidi commented:
<!--
  0 failure: 
  0 warning: 
  2 messages
  
  DangerID: danger-id-default;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>✅  Get a 🏆 for adding description on this PR. - <i>Celebrate on every step!</i></td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>✅ Get a 👍 for requesting reviewers. - <i>Celebrating every win!</i></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7687932fa3f8441905a74d5ef7204b54b3908362
</p>

Weetbix commented on stories/AnchorButton.story.jsx:
```diff
@@ -8,21 +8,56 @@ storiesOf("AnchorButton", module)
       Call to Action
     </AnchorButton>
   ))
+  .add("primary height 56", () => (
```
```suggestion
  .add("primary height large", () => (
```

Weetbix commented on stories/AnchorButton.story.jsx:
```diff
@@ -8,21 +8,56 @@ storiesOf("AnchorButton", module)
+      Call to Action
+    </AnchorButton>
+  ))
+  .add("primary height 32", () => (
```
```suggestion
  .add("primary height small", () => (
```

Weetbix commented on stories/AnchorButton.story.jsx:
```diff
@@ -8,21 +8,56 @@ storiesOf("AnchorButton", module)
   ))
-  .add("tertiary", () => (
-    <AnchorButton href="https://www.vivy.com/" target="_blank" type="tertiary">
+  .add("secondary height 56", () => (
```
```suggestion
 .add("secondary height large", () => (
```

Weetbix commented on stories/AnchorButton.story.jsx:
```diff
@@ -8,21 +8,56 @@ storiesOf("AnchorButton", module)
+      Call to Action
+    </AnchorButton>
+  ))
+  .add("secondary height 32", () => (
```
```suggestion
  .add("secondary height small", () => (
```

Weetbix commented on stories/Button.story.jsx:
```diff
@@ -4,9 +4,33 @@ import { Button, Icon } from "../src/index";
 
 storiesOf("Button", module)
   .add("primary", () => <Button>Call to Action</Button>)
+  .add("primary height 56", () => <Button height="lg">Call to Action</Button>)
```
```suggestion
  .add("primary height large", () => <Button height="lg">Call to Action</Button>)
```

Weetbix commented on stories/Button.story.jsx:
```diff
@@ -4,9 +4,33 @@ import { Button, Icon } from "../src/index";
 storiesOf("Button", module)
   .add("primary", () => <Button>Call to Action</Button>)
+  .add("primary height 56", () => <Button height="lg">Call to Action</Button>)
+  .add("primary height 32", () => <Button height="sm">Call to Action</Button>)
```
```suggestion
  .add("primary height small", () => <Button height="sm">Call to Action</Button>)
```

Weetbix commented on stories/Button.story.jsx:
```diff
@@ -4,9 +4,33 @@ import { Button, Icon } from "../src/index";
   .add("secondary", () => <Button type="secondary">Call to Action</Button>)
-  .add("tertiary", () => <Button type="tertiary">Call to Action</Button>)
-  .add("quaternary", () => <Button type="quaternary">Call to Action</Button>)
+  .add("secondary height 56", () => (
```
```suggestion
  .add("secondary height large", () => (
```

Weetbix commented on stories/Button.story.jsx:
```diff
@@ -4,9 +4,33 @@ import { Button, Icon } from "../src/index";
+      Call to Action
+    </Button>
+  ))
+  .add("secondary height 32", () => (
```
```suggestion
  .add("secondary height small", () => (
```

Weetbix reviewed (changes requested):
Looks good, 

I would say better to name the stories after the attribute names rather than the implemented value, I made some suggestions

paschalidi commented:
> I would say better to name the stories after the attribute names rather than the implemented value, I made some suggestions

Absolutely agree. At the beginning I implemented this such as we parse a number in the component but then decided to change it to string, therefore I forgot to change the story names.. 
Good catch! 

Thanks for the review!

paschalidi committed:
Buttons storries


shots


ADDS SHOTS


prettier


icons


adds Icons


better wording


lock


shots


shots


Icons


cleans


cleans
